velox/exec/RowContainer.cpp

```diff
@@ -109,29 +109,24 @@ RowContainer::RowContainer(
   int32_t firstAggregate = offsets_.size();
   int32_t firstAggregateOffset = offset;
   for (auto& aggregate : aggregates) {
-    // Accumulator offset must be aligned by their alignment size.
-    offset = bits::roundUp(offset, aggregate->accumulatorAlignmentSize());
-    offsets_.push_back(offset);
-    offset += aggregate->accumulatorFixedWidthSize();
     nullOffsets_.push_back(nullOffset);
     ++nullOffset;
     isVariableWidth |= !aggregate->isFixedSize();
     usesExternalMemory_ |= aggregate->accumulatorUsesExternalMemory();
+    VELOX_CHECK_EQ(
+        __builtin_popcount(aggregate->accumulatorAlignmentSize()),
+        1,
+        "Alignment can only be power of 2");
+    alignment_ = std::max(alignment_, aggregate->accumulatorAlignmentSize());
```

Comment:
This code repeats in two places. Consider introducing a helper function to avoid copy-paste.